### PR TITLE
Update cache in background

### LINKED ISSUE #18289: How to refresh the cache every X hours in the background.
.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 13f5f8f9-18dd-9c12-9ed1-fcc6b44ba7fd
* Version Independent ID: 640b4f36-5886-7b69-1c95-cfc8574a02fd
* Content: [Cache in-memory in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/memory?view=aspnetcore-2.1)
* Content Source: [aspnetcore/performance/caching/memory.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/performance/caching/memory.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-performance**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #18289

[Internal review 2.1 version](https://review.docs.microsoft.com/en-us/aspnet/core/performance/caching/memory?view=aspnetcore-2.1&branch=pr-en-us-18295#background-cache-update-1)

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@anurag-oss please review

File: aspnetcore/performance/caching/memory.md [hunk 1]
```diff
@@ -175,6 +175,10 @@ Using a <xref:System.Threading.CancellationTokenSource> allows multiple cache en
 * Use <xref:Microsoft.Extensions.Caching.Memory.ICacheEntry.PostEvictionCallbacks> to set the callbacks that will be fired after the cache entry is evicted from the cache.
 * For most apps, `IMemoryCache` is enabled. For example, calling `AddMvc`, `AddControllersWithViews`, `AddRazorPages`, `AddMvcCore().AddRazorViewEngine`, and many other `Add{Service}` methods in `ConfigureServices`, enables `IMemoryCache`. For apps that are not calling one of the preceding `Add{Service}` methods, it may be necessary to call <xref:Microsoft.Extensions.DependencyInjection.MemoryCacheServiceCollectionExtensions.AddMemoryCache*> in `ConfigureServices`.
 
+## Background cache update
+
+Use as [background service](xref:fundamentals/host/hosted-services) such as <xref:Microsoft.Extensions.Hosting.IHostedService> to update the cache. The background service can recompute the entries and then assign them to the cache only when they’re ready.
```
Comment (anurag-oss) [inline aspnetcore/performance/caching/memory.md:180]: Probable type. "Use a"

File: aspnetcore/performance/caching/memory.md [hunk 2]
```diff
@@ -342,6 +346,10 @@ Using a `CancellationTokenSource` allows multiple cache entries to be evicted as
 
 * Use [PostEvictionCallbacks](/dotnet/api/microsoft.extensions.caching.memory.icacheentry.postevictioncallbacks#Microsoft_Extensions_Caching_Memory_ICacheEntry_PostEvictionCallbacks) to set the callbacks that will be fired after the cache entry is evicted from the cache.
 
+## Background cache update
+
+Use as [background service](xref:fundamentals/host/hosted-services) such as <xref:Microsoft.Extensions.Hosting.IHostedService> to update the cache. The background service can recompute the entries and then assign them to the cache only when they’re ready.
```
Comment (anurag-oss) [inline aspnetcore/performance/caching/memory.md:351]: Probable typo. "Use a"

**Comment by Rick-Anderson** [comment]
@anurag-oss  thanks for the review and original question. Fixed them both.

**Pull request merged**